promdump/promdump.go

```diff
@@ -83,18 +95,42 @@ func init() {
 	}
 }
 
+func getMetricName(metric *promExport) (string, error) {
+
+	if metric.exportName != "" && metric.jobName != "" {
+		return "", errors.New("getMetricName: exportName and jobName are mutually exclusive!  Both detected!")
+	}
+
+	if  metric.exportName != "" {
+		return metric.exportName, nil
+	} else if metric.jobName != "" {
+		return metric.jobName, nil
+	}
+
+	return "", errors.New("getMetricName: no metric name determined!")
+
+}
+
+
 func logMetricCollectorConfig() {
 	// Logs the collector config
 	var collect []string
 	var skip []string
 	for _, v := range collectMetrics {
-		if *out != "" && *out == v.exportName {
-			log.Fatalf("The output file prefix '%v' is reserved. Specify a different --out value.", v.exportName)
+
+		metricName, err := getMetricName(v)
+
+		if err != nil {
+			log.Fatalf("main: %v", err)
+		}
```

Comment:
Should be `logMetricCollectorConfig` in the message rather than `main`.

I generally prefer to have the `if err != nil` immediately below the function whose `err` return we're checking (no extra newline) because my IDE collapses boilerplate checks like if err => fatal.